src/Workspaces/Core/Portable/Workspace/Host/Metadata/IAnalyzerAssemblyLoaderProvider.cs

```diff
@@ -2,9 +2,50 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.IO;
+using Microsoft.CodeAnalysis.Host.Mef;
+
 namespace Microsoft.CodeAnalysis.Host;
 
 internal interface IAnalyzerAssemblyLoaderProvider : IWorkspaceService
 {
-    IAnalyzerAssemblyLoaderInternal GetShadowCopyLoader();
+    IAnalyzerAssemblyLoaderInternal SharedShadowCopyLoader { get; }
 }
+
+/// <summary>
+/// Abstract implementation of an analyzer assembly loader that can be used by VS/VSCode to provide a <see
+/// cref="IAnalyzerAssemblyLoader"/> with an appropriate path.
+/// </summary>
+internal abstract class AbstractAnalyzerAssemblyLoaderProvider : IAnalyzerAssemblyLoaderProvider
+{
+    private readonly ImmutableArray<IAnalyzerAssemblyResolver> _externalResolvers;
+    private readonly Lazy<IAnalyzerAssemblyLoaderInternal> _shadowCopyLoader;
+
+    public AbstractAnalyzerAssemblyLoaderProvider(IEnumerable<IAnalyzerAssemblyResolver> externalResolvers)
+    {
+        _externalResolvers = externalResolvers.ToImmutableArray();
+        _shadowCopyLoader = new(CreateShadowCopyLoader);
+    }
+
+    public IAnalyzerAssemblyLoaderInternal SharedShadowCopyLoader
+        => _shadowCopyLoader.Value;
+
+    private IAnalyzerAssemblyLoaderInternal CreateShadowCopyLoader()
+        => this.WrapLoader(DefaultAnalyzerAssemblyLoader.CreateNonLockingLoader(
+                Path.Combine(Path.GetTempPath(), nameof(Roslyn), "AnalyzerAssemblyLoader"),
```

Comment:
It looks like we're now always using the same path now for VS/OOP - which I think is fine?  I assume in VS we only make shadow copies in the remote workspace now?  So devenv won't shadow copy analyzers and conflict with the remote workspace any more?

Reply:
correct.  every shadow copy loader already puts their work in a unique guid-named subdirectory of the base directory passed in.  So there's no conflicts ever across instances (within the same process, or other process, etc.).  THey are always independent.